andela/ah-shakas
# Ft password reset 160577477

#### What does this PR do?
- Enables a user to reset password using a link sent via email
#### Description of Task to be completed? 
- Users are able to post requests to reset their passwords.
- A token is generated and sent to the user via e mail.
- The token will be used by the user to reset password
- Tests for the functionality are also included.
- Have the following endpoints working
```
POST /api/users/email_sent    body = {"email":email}
```
```
PUT users/password_reset?token= .  body = {"email":email, "password":new_password}
```
#### How should this be manually tested?
- After cloning the repo, set your database variables on .env file.
- Add the following lines to your .env file
```
export EMAIL_HOST=smtp.sendgrid.net
export EMAIL_HOST_USER=shakas
export EMAIL_HOST_PASSWORD=Shakas123
export EMAIL_PORT=587
```
- run the application python manage.py runserver
- test your endpoint.
- NOTE the email you receive has a google url but we are only interested in the token attached to it. 
- from the url in the mail, copy the url from ?token= and paste it on the reset password endpoint to ensure it works.
#### Any background context you want to provide?
- The email sent uses a template obtained from http://leemunroe.github.io/responsive-html-email-template/email.html
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/160577476
